### ISSUE TITLE
[BugFix][MANIFEST] align kw_only flag on signature.params across 34 implemented ops

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#1456]
### Symptom / Motivation

`python scripts/validate_manifest.py --strict` on `main` reports **58 C3 errors** of the form:

```
[ctor] <OpName>: param '<x>' kw_only mismatch — manifest=False, code=True
```

across **34 `status: implemented` ops**. The implementation's `__init__` declares the parameter keyword-only (after a `*,` marker) while the manifest's `signature.params.<x>.kw_only` is `False` (or unset, which defaults to positional).

This is one of the open buckets that must clear before the strict-parity gate (#1291, #1372) can flip to blocking.

### Root Cause Analysis

Drift between two writers of the same contract. Most affected ops are reductions / elementwise / norm — families where `dim`, `keepdim`, `ord`, `dtype`, `unbiased`, `correction`, etc. were tightened to keyword-only in code at some point, while the manifest entry never moved.

Per [.claude/rules/manifest-trust-model.md](.claude/rules/manifest-trust-model.md), the side that disagrees with PyTorch is the side to fix. There is **no mechanical default direction** for this sweep — each `(op, param)` must be evaluated against the PyTorch reference (`torch.<op>` / `torch.nn.functional.<op>` / `torch.linalg.<op>`) and routed accordingly:

| PyTorch reference                          | Fix direction                                                    |
| ------------------------------------------ | ---------------------------------------------------------------- |
| param is after `*` (keyword-only)          | Flip manifest `kw_only: true`. Code is correct.                  |
| param is positional-or-keyword (before `*`) | Move param in code `__init__` before its `*,` marker. Manifest stays `false`. |

A naive "flip everything in manifest to true" approach is **wrong** for this bucket — empirically the families below mostly fall on the second row, not the first. PR #1483 (the first attempt) flipped all 58 sites to manifest-true and was correctly rejected on review for exactly this reason; the second iteration of this issue must do the per-param spot-check before editing.

### Related Files

- `tileops/manifest/elementwise_binary.yaml`
- `tileops/manifest/elementwise_unary_activation.yaml`
- `tileops/manifest/reduction.yaml`
- `tileops/manifest/normalization.yaml`
- `tileops/manifest/scan.yaml`
- `tileops/ops/{activation,norm,reduction,scan,elementwise}/**/*.py` (affected ops' `__init__`)
- `scripts/validate_manifest.py` (`check_c3_ctor_signature_parity`)

## Goal

`python scripts/validate_manifest.py --strict` on `main` reports **zero** `[ctor] … kw_only mismatch` errors, with each `(op, param)` resolution matching the PyTorch reference signature.

## Plan

1. Run `python scripts/validate_manifest.py --strict 2>&1 | grep 'kw_only mismatch'` to enumerate the current 58 sites (34 ops).
2. For each `(op, param)`: open the PyTorch reference signature (interactive: `python -c "import inspect, torch; print(inspect.signature(torch.<op>))"`). Decide based on the table in Root Cause Analysis:
   - If PyTorch has the param **after `*`** → flip manifest `kw_only: true` (use round-trip `ruamel.yaml` to preserve comments / key order).
   - If PyTorch has the param **before `*`** → in the op's `__init__`, move the param out from after the `*,` marker so it becomes positional-or-keyword. Manifest stays as-is.
3. Both kinds of edits land in this PR; constraint is that the two sides finish aligned (validator green) after one cohesive change set.
4. If a family-by-family split produces cleaner reviews than one mega-PR, split. Either is fine.
5. Re-run `--strict`; confirm `kw_only mismatch` count is 0; commit.

Empirically observed routing for the affected families (from PR #1483 review by `Ibuki-wind`, sha `1ea4c2e`):

| Family                                      | Direction                                              |
| ------------------------------------------- | ------------------------------------------------------ |
| in-place activation `inplace` (Relu/Silu/…) | code → positional-or-keyword (PyTorch is positional)   |
| `BatchNormFwdOp.{training,momentum,eps}`    | code → positional-or-keyword                           |
| reductions `dim` / `keepdim` (Softmax/Sum/Mean/Amax/Amin/Prod/Argmax/Argmin/All/Any/CountNonzero/LogSumExp/Var/Std/VarMean) | code → positional-or-keyword |
| `linalg.vector_norm`-family `ord` / `dim`   | code → positional-or-keyword                           |
| `Cumsum/CumprodFwdOp.dim`                   | code → positional-or-keyword                           |
| `AddFwdOp.alpha`, `SubFwdOp.alpha`, `DivFwdOp.rounding_mode` | manifest → `kw_only: true` (PyTorch has them after `*`) |
| layer-norm-family / rms-norm-family kw-only params (if flipped) | manifest → `kw_only: true` |

This table is the empirical starting point — re-verify each entry against PyTorch before editing.

Affected ops (34): AddFwdOp, AllFwdOp, AmaxFwdOp, AminFwdOp, AnyFwdOp, ArgmaxFwdOp, ArgminFwdOp, BatchNormFwdOp, CountNonzeroFwdOp, CumprodFwdOp, CumsumFwdOp, DivFwdOp, EluFwdOp, HardsigmoidFwdOp, HardswishFwdOp, HardtanhFwdOp, InfNormFwdOp, L1NormFwdOp, L2NormFwdOp, LeakyReluFwdOp, LogSoftmaxFwdOp, LogSumExpFwdOp, MeanFwdOp, MishFwdOp, ProdFwdOp, ReluFwdOp, SeluFwdOp, SiluFwdOp, SoftmaxFwdOp, StdFwdOp, SubFwdOp, SumFwdOp, VarFwdOp, VarMeanFwdOp.

## Constraints

- Joint manifest+impl PR for this sweep. Code `__init__` edits are explicitly in scope — moving params across the `*,` marker is the correct fix when PyTorch has them positional. A previous formulation forbade code-side edits; that constraint was lifted after PR #1483's review showed most ops need exactly that direction.
- Per [.claude/rules/manifest-trust-model.md](.claude/rules/manifest-trust-model.md) "PyTorch parity" rule: **the side that disagrees with PyTorch is the side to fix.** No mechanical default direction; spot-check each `(op, param)` against the live PyTorch signature.
- Do NOT use `parity_opt_out` — the field was removed by #1292.
- Do NOT demote ops to `status: spec-only` to silence the validator. The implementations are present; only the kw_only declaration drifted.

## Acceptance Criteria

- [ ] Modified files pass unit tests (including the affected per-family tests under `tests/ops/{activation,norm,reduction,scan}/`).
- [ ] AC-1: `python scripts/validate_manifest.py --strict` reports 0 `[ctor] … kw_only mismatch` errors.
- [ ] AC-2: No op is demoted from `status: implemented` to `status: spec-only` as part of this fix.
- [ ] AC-3: All 34 ops listed in the Plan section are addressed.
- [ ] AC-4: Linked from meta #1372 as a prerequisite.
- [ ] AC-5: Each `(op, param)` resolution direction matches the PyTorch reference signature; the PR body documents per-family direction so reviewers can spot-check without re-deriving.

**Comment by lcy-seso** [comment]
nightshift pipeline gave up on this issue.

**Reason**: Bot reviewer (Ibuki-wind) issued CHANGES_REQUESTED with 4 substantive blockers (PR #1483, head 1ea4c2e) demanding strict PyTorch-API parity: revert kw_only on inplace activation params, batch_norm training/momentum/eps, softmax/sum/mean/amax/argmax/all/any/count_nonzero/vector_norm dim/keepdim, and cumsum/cumprod dim. Per .claude/rules/manifest-trust-model.md the side that disagrees with PyTorch is the side to fix — meaning the corresponding code __init__ signatures across ~20 ops in tileops/ops/{activation,norm,reduction,scan}/ would need to be made positional-or-keyword too. This contradicts the issue body's directive ('default direction is manifest -> kw_only: true') and exceeds AC-1's intended scope (manifest-only edits). Needs human policy call between: (a) revise issue scope to include code __init__ changes; (b) reply rejecting bot review citing the issue author's explicit override; (c) split into per-family follow-ups. Escalating rather than autonomously deciding.
**PR**: #1483 (head `1ea4c2e`, CHANGES_REQUESTED by Ibuki-wind)
**Run dir**: `/home/cy/TileOps-dev/TileOPs/.foundry/runs/issue-1456/pipeline` (logs under `/home/cy/TileOps-dev/TileOPs/.foundry/runs/issue-1456/pipeline/../logs/`)
**Action**: label flipped `nightshift` → `nightshift-blocked`. Investigate, fix the underlying problem (decide policy between issue-body directive vs strict PyTorch parity), and either re-add the `nightshift` label (re-queue) or close this issue.

**Comment by lcy-seso** [comment]
Resolved by PR #1483 (squash-merged into testbed at d50ec06d).
